schemas/string_interpolation/transforms/generate_date_range.json

```diff
@@ -0,0 +1,74 @@
+{
+  "$id": "https://eq.ons.gov.uk/string_interpolation/transforms/generate_date_range.json",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "generate_date_range": {
```

Comment:
Could just be called `date_range`, to me, `generate` doesn't add much plus proposal for date range in the new rules structure uses `date-range` which placeholders would eventually be refactored to use.

Reply:
`date_bounds`?

Reply:
Not sure if that is very clear. Don't mind that, maybe see what others think. Maybe `range` would help, even if bounds implies it is a range, i.e `date_range_bounds`?. If we chose `date_bounds` or `date_range_bounds`, the format transform should be updated accordingly, i.e `format_date_bounds` etc.

Reply:
We also have a transform called `calculate_date_difference` - I just followed existing naming patterns where I could

Reply:
Yeah `date_range_bounds` is most descriptive, it's the boundary dates for a date range.